network/p2p/p2ptest/client.go

```diff
@@ -19,72 +20,112 @@ import (
 		// Send the request asynchronously to avoid deadlock when the server
 		// sends the response back to the client
-		go func() {
-			require.NoError(t, serverNetwork.AppGossip(ctx, clientNodeID, gossipBytes))
-		}()
+		for nodeID := range peers {
+			go func() {
+				require.NoError(t, peerNetworks[nodeID].AppGossip(ctx, nodeID, gossipBytes))
+			}()
+		}
 
 		return nil
 	}
 
-	clientSender.SendAppRequestF = func(ctx context.Context, _ set.Set[ids.NodeID], requestID uint32, requestBytes []byte) error {
-		// Send the request asynchronously to avoid deadlock when the server
-		// sends the response back to the client
-		go func() {
-			require.NoError(t, serverNetwork.AppRequest(ctx, clientNodeID, requestID, time.Time{}, requestBytes))
-		}()
+	peerSenders[clientNodeID].SendAppRequestF = func(ctx context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, requestBytes []byte) error {
+		for nodeID := range nodeIDs {
+			network, ok := peerNetworks[nodeID]
+			if !ok {
+				return fmt.Errorf("%s is not connected", nodeID)
+			}
+
+			// Send the request asynchronously to avoid deadlock when the server
+			// sends the response back to the client
+			go func() {
+				require.NoError(t, network.AppRequest(ctx, clientNodeID, requestID, time.Time{}, requestBytes))
+			}()
+		}
 
 		return nil
 	}
 
-	serverSender.SendAppResponseF = func(ctx context.Context, _ ids.NodeID, requestID uint32, responseBytes []byte) error {
-		// Send the request asynchronously to avoid deadlock when the server
-		// sends the response back to the client
-		go func() {
-			require.NoError(t, clientNetwork.AppResponse(ctx, serverNodeID, requestID, responseBytes))
-		}()
+	for nodeID := range peers {
+		peerSenders[nodeID].SendAppResponseF = func(ctx context.Context, _ ids.NodeID, requestID uint32, responseBytes []byte) error {
+			// Send the request asynchronously to avoid deadlock when the server
+			// sends the response back to the client
+			go func() {
+				require.NoError(t, peerNetworks[clientNodeID].AppResponse(ctx, nodeID, requestID, responseBytes))
+			}()
 
-		return nil
+			return nil
+		}
 	}
 
-	serverSender.SendAppErrorF = func(ctx context.Context, _ ids.NodeID, requestID uint32, errorCode int32, errorMessage string) error {
-		// Send the request asynchronously to avoid deadlock when the server
-		// sends the response back to the client
-		go func() {
-			require.NoError(t, clientNetwork.AppRequestFailed(ctx, serverNodeID, requestID, &common.AppError{
-				Code:    errorCode,
-				Message: errorMessage,
-			}))
-		}()
-
-		return nil
+	for nodeID := range peers {
+		peerSenders[nodeID].SendAppErrorF = func(ctx context.Context, _ ids.NodeID, requestID uint32, errorCode int32, errorMessage string) error {
+			// Send the request asynchronously to avoid deadlock when the server
+			// sends the response back to the client
+			nodeIDCopy := nodeID
```

Comment:
No longer needed with go 1.22 (https://go.dev/blog/loopvar-preview), but it looks like this shadows the loop variable incorrectly (according to < go1.22 recommended pattern to shadow loop variable).

The closure should capture `nodeID` and perform the copy when the function is invoked at which point `nodeID` can be overwritten by the loop.

Could we either remove the copy and rely on the fact that we're running > go1.22 or move the copy outside of the closure?

Needed to convince myself this was the case since I wasn't sure how golang would handle this case, so I wrote this program and get the following results:

```golang
package main

import (
	"fmt"
)

func main() {
	peers := make(map[int]struct{})
	peers[1] = struct{}{}
	peers[2] = struct{}{}
	peers[3] = struct{}{}

	funcs := make([]func() int, 0)
	for nodeID := range peers {
		funcs = append(funcs, func() int {
			nodeIDCopy := nodeID
			return nodeIDCopy
		})
	}

	for _, f := range funcs {
		fmt.Println(f())
	}
}
```

go1.21 result:
```
3
3
3
```

go1.22 result:
```
1
2
3
```

Reply:
Removed the copy var